lib/src/internal/utils/utils.dart

```diff
@@ -0,0 +1,41 @@
+import 'dart:io';
+import 'dart:isolate';
+
+import 'package:yaml/yaml.dart';
+
+// Extract the package version number from
+// the pubspec.yaml file if it's found
+Future<String?> findPackageVersion() async {
+  File? pubspecFile = await findPubspecFile();
+  if (pubspecFile != null) {
+    Map pubspec = loadYaml(pubspecFile.readAsStringSync());
```

Comment:
what you are doing is probably fine, but just curious if you have seen [this](https://pub.dev/documentation/pubspec_parse/latest/pubspec_parse/Pubspec/Pubspec.parse.html)

Reply:
I think that one converts yaml string to a pubspec file